### PR TITLE
Add mock linting

### PR DESCRIPTION
**Pull request by dcramer**
This is inspired by http://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html

### CODE REVIEW

File: src/sentry/lint/mock_check.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import
+
+import ast
+
+
+class MockAttrVisitor(ast.NodeVisitor):
+    non_existent_methods = [
```
Comment (mattrobenolt) [inline src/sentry/lint/mock_check.py]: Should do a `frozenset`. :icecream:

**Comment by mattrobenolt** [comment]
Being lazy, but what happens if we had a method called `not_called` on a non-Mock object, for example? Will that be rejected too?

**Comment by dcramer** [comment]
@mattrobenolt yep, you cant do that now

**Comment by mattrobenolt** [comment]
Would it make sense to restrict this to just `tests/*` files then? Or is it not a big enough deal?

I guess I'm saying it's technically possible for this to yield false positives on things that aren't Mock objects.

**Comment by dcramer** [comment]
@mattrobenolt dont call functions things that look like mock objects, problem solved

**Comment by mattrobenolt** [comment]
:trollface:

**Pull request merged**